receiver/oktareceiver/logs_test.go

```diff
@@ -0,0 +1,272 @@
+			require.Contains(t, req.URL.String(), "since=")
+			require.Contains(t, req.URL.String(), "limit="+strconv.Itoa(oktaMaxLimit))
+			if strings.Contains(req.URL.String(), "after=") {
+				return mockAPIResponseTooManyRequests(), nil
+			}
+			return mockAPIResponseOK1000Logs(), nil
+		},
+	}
+
+	err = recv.poll(context.Background())
+	require.NoError(t, err)
+
+	logs := sink.AllLogs()
+
+	require.Equal(t, 1000, logs[0].ResourceLogs().At(0).ScopeLogs().At(0).LogRecords().Len())
+	require.Equal(t, mockNextURL, recv.nextURL)
+
+	err = recv.Shutdown(context.Background())
+	require.NoError(t, err)
+}
+
+func TestPollOverflow(t *testing.T) {
+	cfg := createDefaultConfig().(*Config)
+	mockDomain := "observiq.okta.com"
+	cfg.Domain = mockDomain
+
+	sink := &consumertest.LogsSink{}
+	recv, err := newOktaLogsReceiver(cfg, zap.NewNop(), sink)
+	require.NoError(t, err)
+
+	recv.client = &mockHTTPClient{
+		mockDo: func(req *http.Request) (*http.Response, error) {
+			require.Contains(t, req.URL.String(), "since=")
+			require.Contains(t, req.URL.String(), "limit="+strconv.Itoa(oktaMaxLimit))
+			if strings.Contains(req.URL.String(), "after=") {
+				return mockAPIResponseOKBasic(), nil
+			}
+			return mockAPIResponseOK1000Logs(), nil
+		},
+	}
+
+	err = recv.poll(context.Background())
+	require.NoError(t, err)
+
+	logs := sink.AllLogs()
+
+	require.Equal(t, 1002, logs[0].ResourceLogs().At(0).ScopeLogs().At(0).LogRecords().Len())
+	require.Equal(t, mockNextURL, recv.nextURL)
+
+	err = recv.Shutdown(context.Background())
+	require.NoError(t, err)
+}
+
+func TestPollPublishedAfterPollTime(t *testing.T) {
+	cfg := createDefaultConfig().(*Config)
+	mockDomain := "observiq.okta.com"
+	cfg.Domain = mockDomain
+
+	sink := &consumertest.LogsSink{}
+	recv, err := newOktaLogsReceiver(cfg, zap.NewNop(), sink)
+	require.NoError(t, err)
+
+	recv.client = &mockHTTPClient{
+		mockDo: func(req *http.Request) (*http.Response, error) {
+			require.Contains(t, req.URL.String(), "since=")
+			require.Contains(t, req.URL.String(), "limit="+strconv.Itoa(oktaMaxLimit))
+			if strings.Contains(req.URL.String(), "after=") {
+				return mockAPIResponseOKBasic(), nil
+			}
+			return mockAPIResponseOK1000LogsAfter(), nil
+		},
+	}
+
+	err = recv.poll(context.Background())
+	require.NoError(t, err)
+
+	logs := sink.AllLogs()
+
+	require.Equal(t, 1000, logs[0].ResourceLogs().At(0).ScopeLogs().At(0).LogRecords().Len())
+	require.Equal(t, mockNextURL, recv.nextURL)
+
+	err = recv.Shutdown(context.Background())
+	require.NoError(t, err)
+}
+
+func newReceiver(t *testing.T, cfg *Config, c consumer.Logs) *oktaLogsReceiver {
+	r, err := newOktaLogsReceiver(cfg, zap.NewNop(), c)
+	require.NoError(t, err)
+	return r
+}
+
+func jsonFileAsPlogs(filepath string) (plog.Logs, error) {
+	expectedFileBytes, err := os.ReadFile(filepath)
+	if err != nil {
+		return plog.Logs{}, nil
+	}
+	unmarshaler := &plog.JSONUnmarshaler{}
+	return unmarshaler.UnmarshalLogs(expectedFileBytes)
+}
```

Comment:
nit: You can use this package to read/write logs for tests: 
https://pkg.go.dev/github.com/open-telemetry/opentelemetry-collector-contrib/pkg/golden

Reply:
Ooo good tip, thats way cleaner